src/AttachVS/AttachVs.cs

```diff
@@ -215,61 +223,51 @@ private static bool IsVsOrNull(Process process)
         return isVs;
     }
 
-    private static bool IsCorrectParent(Process currentProcess, Process parent)
+    private static Process? GetParentProcess(Process process)
     {
-        try
+        int id = GetProcessId(process);
+        if (id != -1)
         {
-            // Parent needs to start before the child, otherwise it might be a different process 
-            // that is just reusing the same PID.
-            if (parent.StartTime <= currentProcess.StartTime)
+            try
             {
-                return true;
-            }
-
-            Trace($"Process {parent.ProcessName} ({parent.Id}) is not a valid parent because it started after the current process.");
-            return false;
-        }
-        catch
-        {
-            // Access denied or process exited while we were holding the Process object.
-            return false;
-        }
-    }
-
-    private static Process GetParentProcess(Process process)
-    {
-        int id;
-        try
-        {
-            var handle = process.Handle;
-            var res = NtQueryInformationProcess(handle, 0, out var pbi, Marshal.SizeOf<PROCESS_BASIC_INFORMATION>(), out int size);
+                Process parent = Process.GetProcessById(id);
 
-            var p = res != 0 ? -1 : pbi.InheritedFromUniqueProcessId.ToInt32();
+                // Parent needs to start before the child, otherwise it might be a different process
+                // that is just reusing the same PID.
+                if (parent.StartTime <= process.StartTime)
+                {
+                    return parent;
+                }
 
-            id = p;
-        }
-        catch
-        {
-            id = -1;
+                Trace($"Process {parent.ProcessName} ({parent.Id}) is not a valid parent because it started after the current process.");
+            }
+            catch
+            {
+                // throws when parent no longer runs
+            }
         }
 
-        Process parent = null;
-        if (id != -1)
+        return null;
+
+        static int GetProcessId(Process process)
         {
             try
             {
-                parent = Process.GetProcessById(id);
+                var handle = process.Handle;
+                var res = NtQueryInformationProcess(handle, 0, out var pbi, Marshal.SizeOf<PROCESS_BASIC_INFORMATION>(), out int size);
+
+                var p = res != 0 ? -1 : pbi.InheritedFromUniqueProcessId.ToInt32();
+
+                return p;
             }
             catch
             {
-                // throws when parent no longer runs
+                return -1;
             }
         }
-
-        return IsCorrectParent(process, parent) ? parent : null;
```

Comment:
Why did this logic have to move into the body of the other method directly? Separating it into a named method seemed clearer.

Reply:
Seems more complex to have all the logic split but I will revert.

Separated looks like:

![image](https://user-images.githubusercontent.com/11340282/170516748-cca05099-0993-4a05-aff5-716412e3a3d4.png)

Merged:

![image](https://user-images.githubusercontent.com/11340282/170516902-07a24e8f-0ffa-43f5-a883-fcb9e5782cca.png)